llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -1750,6 +1752,97 @@ void lowerGlobalsToVector(Module &M) {
 
 } // namespace
 
+static void checkSLMInit(Module &M) {
+  SmallPtrSet<const Function *, 8u> SLMInitKernels;
+  SmallPtrSet<const Function *, 8u> LocalAccessorKernels;
+
+  for (auto &F : M) {
+    if (!isSlmInit(F)) {
+      bool LocalAccessorUsed = false;
```

Comment:
This !isSLMInit(F) branch looks a bit too complicated. It can be simple and more robust IMO:
The search of loca_accessors can be done only in parameters of ESIMD kernels, I.e.:
```C++
    if (!esimd::isESIMDKernel(F))
      continue;
   for (const Argument &Arg : F.args()) {
     if (Arg is a local_acc)
       LocalAccessorKernels.insert(F);
   }
```
Here there is no need in graph traversal-up, no need in checking args of every function, no need in SPIRV_LOCAL_ACCESSOR_PREF

Reply:
The other/else branch (if isSLMInit(F) part of the code) looks great.